lib/apimatic-core/types/parameter.rb

```diff
@@ -88,34 +88,24 @@ def get_default_content_type
       @default_content_type
     end
 
-    # Template setter in case of oneOf or anyOf.
-    # @param [String] template Template for the parameter.
-    # @return [Parameter] An updated instance of Parameter.
-    def template(template)
-      @template = template
+    # Setter for the validator.
+    # @param validator [callable] The validator function to be set.
+    # @return [Parameter] An updated instance of the Parameter class.
+    def validator(validator)
+      @validator = validator
       self
     end
 
-    # Template getter in case a template is set.
-    # @return [Parameter] Returns template for the parameter.
-    def get_template
-      @template
-    end
-
     # Validates the parameter value to be sent in the request.
-    # @raise [ValueError] The value error if the parameter is required but the value is nil.
+    # @raise [ArgumentError] If the parameter is required but the value is nil.
```

Comment:
Nit, Non-blocker: This return docstring is not needed.

Reply:
Done